server/src/main/java/org/opensearch/action/search/SearchRequestListenerManager.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * SearchRequestListenerManager manages listeners registered to search requests,
+ * and is responsible for creating the {@link SearchRequestOperationsListener.CompositeListener}
+ * with the all listeners enabled at cluster-level and request-level.
+ *
+ *
+ * @opensearch.internal
+ */
+public class SearchRequestListenerManager {
```

Comment:
Naming nitpick... "Manager" is definitely an overused term, but this class doesn't really manage anything. It's really just a collection of `SearchRequestOperationsListener` instances with a function to build a composite listener from that collection. I might just name this the plural `SearchRequestOperationsListeners` (see `org.opensearch.index.shard.{RefreshListeners,GlobalCheckpointListeners}` for an example of this naming pattern.